plugins/modules/k8s_drain.py

```diff
@@ -0,0 +1,415 @@
+#!/usr/bin/python
+# -*- coding: utf-8 -*-
+
+# Copyright (c) 2021, Aubin Bikouo <@abikouo>
+# GNU General Public License v3.0+ (see COPYING or https://www.gnu.org/licenses/gpl-3.0.txt)
+
+from __future__ import absolute_import, division, print_function
+
+__metaclass__ = type
+
+
+DOCUMENTATION = r'''
+
+module: k8s_drain
+
+short_description: Drain, Cordon, or Uncordon node in k8s cluster.
+
+author: Aubin Bikouo (@abikouo)
+
+description:
+  - Drain node in preparation for maintenance same as kubectl drain.
+  - Cordon will mark node as unschedulable.
+  - Uncordon will mark node as schedulable.
+  - The given node will be marked unschedulable to prevent new pods from arriving.
+  - Then drain deletes all pods except mirror pods (which cannot be deleted through the API server).
+
+extends_documentation_fragment:
+  - kubernetes.core.k8s_auth_options
+
+options:
+  state:
+    description:
+    - Determines whether to drain, cordon, or uncordon node.
```

Comment:
Can we name this parameter something else, since `state` is known for values like `present` and `absent`? Something like `operation`.

Reply:
the ``state`` is used for some modules like the ``ansible.builtin.file`` module with value different from ``present`` and ``absent``, but here of course this can be changed if confusing